### PR TITLE
Add extern C to public headers

### PR DESCRIPTION
**Pull request by studoret**
Note that I placed '#ifdef __cplusplus' after '#include' directives because I consider that each header file takes care of itself.
By the way, Happy new year !

### CODE REVIEW

File: include/upipe/udict.h [hunk 1]
```diff
@@ -596,7 +596,7 @@ static inline int udict_set_opaque_from_hex(struct udict *udict,
 {
     size_t size = (strlen(value) + 1) / 2;
     uint8_t v[size];
-    int i;
+    uint32_t i;
```
Comment (quarium) [inline include/upipe/udict.h]: Maybe size_t is more appropriate

Comment (studoret) [reply include/upipe/udict.h]: I agree.

**Comment by nto** [comment]
Could you squash the two sign-compare commits into a single one?

**Comment by studoret** [comment]
Done

**Comment by nto** [comment]
Thanks!

**Pull request merged**